homeassistant/components/zha/light.py

```diff
@@ -130,7 +129,9 @@ def brightness(self):
     @property
     def device_state_attributes(self):
         """Return state attributes."""
-        return self.state_attributes
+        state_attrs = self.state_attributes.copy()
+        state_attrs["off_brightness"] = self._off_brightness
+        return state_attrs
```

Comment:
IMO we're doing it wrong. We shouldn't return `self.state_attributes`. Just return a dict with `off_brigthness` key. However it is a bit grey area, as this "attribute" really doesn't describe the "current state".

Reply:
But we were returning self.state_attributes before. Was that wrong?
Do you have any other suggestion on how to save the attribute?

Reply:
Yeah, I'm 75% sure that was wrong and we shouldn't continue doint it. `state_attributes` are implemented by component base class, platform class shouldn't mess with it. `device_state_attributes` are meant to be implemented by the platform classes https://github.com/home-assistant/home-assistant/blob/62adff23f935cbb644c8dfb17e8e109f407374a8/homeassistant/helpers/entity.py#L156 and line 150 we should do it like deConz does it https://github.com/home-assistant/home-assistant/blob/f6be61c6b75a860384fe75e1b0ef9b801139c2c8/homeassistant/components/deconz/light.py#L190-L196